main.go

```diff
@@ -296,9 +316,82 @@ func (s XcodeArchiveStep) ProcessInputs() (Config, error) {
 		config.ArtifactName = productName
 	}
 
+	codesignManager, err := s.createCodesignManager(config)
+	if err != nil {
+		return Config{}, err
+	}
+	config.CodesignManager = codesignManager
+
 	return config, nil
 }
 
+func (s XcodeArchiveStep) createCodesignManager(config Config) (codesign.Manager, error) {
+	var authType codesign.AuthType
+	switch config.CodeSigningAuthSource {
+	case codeSignSourceOff:
+		authType = codesign.NoAuth
```

Comment:
Does it make sense to continue this case?

Reply:
Please take a look now, we made it a bit simpler in the case of `codesign.NoAuth`